services-core/src/solution_submission.rs

```diff
@@ -87,9 +87,9 @@ impl From<Error> for SolutionSubmissionError {
     }
 }
 
-pub struct StableXSolutionSubmitter {
+pub struct StableXSolutionSubmitter<T = RetryWithGasPriceIncrease> {
     contract: Arc<dyn StableXContract>,
-    retry_with_gas_price_increase: Box<dyn SolutionTransactionSending>,
+    retry_with_gas_price_increase: T,
```

Comment:
This can't be a Box<dyn> anymore  because the trait is a generic function.

Reply:
`dyn Trait` is a trait object. only "object safe" traits can be converted to trait objects but Retry isn't object safe because it has a method with generics: https://doc.rust-lang.org/reference/items/traits.html#object-safety